scripts/tf_cnn_benchmarks/benchmark_cnn.py

```diff
@@ -208,7 +208,7 @@
                     'If specified, after the graph has been partitioned and '
                     'optimized, write out each partitioned graph to a file '
                     'with the given prefix.')
-flags.DEFINE_string('optimizer', 'sgd',
+flags.DEFINE_enum('optimizer', 'sgd', ('momentum', 'sgd', 'rmsprop'),
                     'Optimizer to use: momentum or sgd or rmsprop')
 flags.DEFINE_float('init_learning_rate', None,
                    'Initial learning rate for training.')
```

Comment:
Line 389 has the `'variable_update'` flag. Can you also change that to an enum? The possible values are `'parameter_server'`, `'replicated'`, `'distributed_replicated'`, `'independent'`, `'distributed_all_reduce'`, and `'horovod'`.